editor/uiclasses.js

```diff
@@ -232,4 +314,54 @@ class OutputNode extends Node {
 
         return this.inputs[0].connection.parent.getFormula();
     }
+
+    getAsymptotes(xMin, xMax) {
+        return this.inputs[0].connection.parent.getAsymptotes(xMin, xMax);
+    }
+}
+
+
+function nthSineZero(func, x) {
+    // floor((f(x) - sin^-1(0)) / pi) = floor(f(x) / pi)
+    return Math.floor((Number(nerdamer(func).evaluate({"x": x})) / Math.PI));
+}
+
+
+/**
+ * The floor of the nth zero at the value x for cosine. Used for finding the nth tangent asymptote at x = ?
+ * @param func The inner function to find the zero of.
+ * @param x The x-value to find the zero at.
+ */
+function nthCosineZero(func, x) {
+    // floor((f(x) - cos^-1(0)) / pi)
+    return Math.floor((Number(nerdamer(func).evaluate({"x": x}) - Math.acos(0)) / Math.PI));
+}
+
+
+function getDerivative(funcStr) {
```

Comment:
lets go derivatives lol

Reply:
the function is unused - will not be unused later